x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -33,94 +32,142 @@ func init() {
-		ResponseFormat string `config:"sql_response_format"`
-	}{ResponseFormat: tableResponseFormat}
+	b := &MetricSet{BaseMetricSet: base}
 
-	if err := base.Module().UnpackConfig(&config); err != nil {
+	if err := base.Module().UnpackConfig(&b.Config); err != nil {
 		return nil, err
 	}
 
-	if config.ResponseFormat != variableResponseFormat && config.ResponseFormat != tableResponseFormat {
-		return nil, fmt.Errorf("invalid sql_response_format value: %s", config.ResponseFormat)
+	if b.Config.ResponseFormat != "" && b.Config.ResponseFormat != variableResponseFormat && b.Config.ResponseFormat != tableResponseFormat {
+		return nil, fmt.Errorf("invalid sql_response_format value: %s", b.Config.ResponseFormat)
+	}
+
+	for _, q := range b.Config.Queries {
+		if q.ResponseFormat != variableResponseFormat && q.ResponseFormat != tableResponseFormat {
+			return nil, fmt.Errorf("invalid sql_response_format value: %s", q.ResponseFormat)
+		}
 	}
 
-	return &MetricSet{
-		BaseMetricSet:  base,
-		Driver:         config.Driver,
-		Query:          config.Query,
-		ResponseFormat: config.ResponseFormat,
-	}, nil
+	if b.Config.Query == "" && len(b.Config.Queries) == 0 {
+		return nil, fmt.Errorf("no query input provided, must provide either sql_query or sql_queries")
+	}
+
+	if b.Config.Query != "" && len(b.Config.Queries) > 0 {
+		return nil, fmt.Errorf("both query inputs provided, must provide either sql_query or sql_queries")
+	}
+
+	return b, nil
 }
 
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
 // It calls m.fetchTableMode() or m.fetchVariableMode() depending on the response
 // format of the query.
-func (m *MetricSet) Fetch(ctx context.Context, report mb.ReporterV2) error {
-	db, err := sql.NewDBClient(m.Driver, m.HostData().URI, m.Logger())
+func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+	db, err := sql.NewDBClient(m.Config.Driver, m.HostData().URI, m.Logger())
 	if err != nil {
-		return errors.Wrap(err, "error opening connection")
+		return fmt.Errorf("could not open connection: %w", err)
 	}
 	defer db.Close()
 
-	if m.ResponseFormat == tableResponseFormat {
-		mss, err := db.FetchTableMode(ctx, m.Query)
-		if err != nil {
-			return err
-		}
-
-		for _, ms := range mss {
-			report.Event(m.getEvent(ms))
-		}
-
-		return nil
+	queries := m.Config.Queries
+	if len(queries) == 0 {
+		one_query := query{Query: m.Config.Query, ResponseFormat: m.Config.ResponseFormat}
+		queries = append(queries, one_query)
 	}
 
-	ms, err := db.FetchVariableMode(ctx, m.Query)
-	if err != nil {
-		return err
+	for _, q := range queries {
+		if q.ResponseFormat == tableResponseFormat {
+			// Table format
+			mss, err := db.FetchTableMode(ctx, q.Query)
+			if err != nil {
+				return err
```

Comment:
We are trying to avoid non-contextualized error returns in Beats so please, add some context to this `err` return. The context here should indicate that the user-faced `table mode` has failed its execution somehow.

Reply:
Updated to added context returns.